components/notion/actions/common.mjs

```diff
@@ -0,0 +1,220 @@
+/**
+ * Builds Notion text property for Notion blocks
+ * @param {string} content - The text content
+ * @returns
+ */
+function buildTextProperty(content) {
+  return [
+    {
+      type: "text",
+      text: {
+        content,
+      },
+    },
+  ];
+}
+
+/**
+ * Builds a Notion block object
+ * @param {string} type - The block type
+ * @param {list} propList - A list of block object values for the block type, in the following format:
+ *  [
+ *    {
+ *      label: text,
+ *      value: "some text",
+ *    },
+ *    {
+ *      label: checked,
+ *      value: true,
+ *    }
+ *  ]
+ */
+function buildBlock(type, propList) {
+  const blockProps = {};
+  for (const { label, value } of propList) {
+    if (label == "text") {
+      blockProps[label] = buildTextProperty(value);
+    } else {
+      blockProps[label] = value;
+    }
+  }
+  return {
+    object: "block",
+    type,
+    [type]: {
+      ...blockProps,
+    },
+  }
+}
+
+/**
+ * Notion block types, containing:
+ *   - Notion key string
+ *   - props
+ *   - additionalProps
+ */
+const blockType = {
```

Comment:
this can be a constant I guess in `notion/common/constants.mjs` for instance

Reply:
Addressed